app/models/state_changer/tube_base.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module StateChanger
+  # Shared behaviour for tubes
+  class TubeBase < StateChanger::Base
+    # Updates the state of the labware to the target state.  The basic implementation does this by updating
+    # all of the TransferRequest instances to the state specified.
+    # @return [Void]
+    def update_labware_state
+      update_associated_requests
```

Comment:
this method isn't implemented, so this class shouldn't be used on its own?

Reply:
Yep, added documentation and explicit 'raise NotImplementedError'